mergify_engine/tests/unit/test_json.py

```diff
@@ -0,0 +1,51 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2018–2020 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import enum
+import json
+
+from mergify_engine import json as mergify_json
+
+
+class Colour(enum.Enum):
+    RED = 1
+    GREEN = 2
+    BLUE = 3
+
+
+with_enum = {
+    "name": "hello",
+    "conditions": [],
+    "actions": {"merge": {"strict": Colour.BLUE}},
+}
+
+with_enum_encode = {
+    "name": "hello",
+    "conditions": [],
+    "actions": {
+        "merge": {"strict": {"__pytype__": "enum", "class": "Colour", "name": "BLUE"}}
+    },
+}
+
+
+def test_encode_enum():
+    mergify_json.register_type(Colour)
+    assert mergify_json.dumps(with_enum) == json.dumps(with_enum_encode)
```

Comment:
I'm not sure `json.dumps` is supposedly idempotent so that might work by luck also.
Rather do:
```suggestion
    assert json.loads(mergify_json.dumps(with_enum)) == with_enum_encode
```

Reply:
ok